store/src/java/com/zimbra/cs/service/account/Auth.java

```diff
@@ -508,4 +528,36 @@ public static void addAccountToLogContextByAuthToken(Provisioning prov, AuthToke
         if (aid != null && !aid.equals(id))
             AccountUtil.addAccountToLogContext(prov, aid, ZimbraLog.C_ANAME, ZimbraLog.C_AID, null);
     }
+
+    public static AuthToken validateAndCreateNewJwtToken(String jwt, Account acct, Provisioning prov) throws ServiceException {
+        if (StringUtil.isNullOrEmpty(jwt)) {
+            throw ServiceException.AUTH_REQUIRED();
+        }
+        String tokenKey = acct.getJWTKey();
```

Comment:
This has been taken care under zcs-2398 and tested basic(sending tokenType="JWT" and jwtAuthToken in req) scenario through soap.